test/test_enzyme.jl

```diff
@@ -183,54 +184,69 @@ end
     fill_halo_regions!(u)
     fill_halo_regions!(v)
 
-    @inline function tracer_flux(x, y, t, c, p)
+    @inline function tracer_flux(i, j, grid, clock, model_fields, p)
         c₀ = p.surface_tracer_concentration
         u★ = p.piston_velocity
-        return - u★ * (c₀ - c)
+        return - u★ * (c₀ - model_fields.c[i, j, p.level])
```

Comment:
```suggestion
        return - u★ * (c₀ - model_fields.c[i, j, grid.Nz])
```

will work too if this is a surface flux.